sparkly/utils.py

```diff
@@ -59,6 +64,89 @@ def absolute_path(file_path, *rel_path):
     )
 
 
+def chunks(seq, n):
+    """Iterate over seq by chunks of n elements.
+
+    >>> list(chunks([1, 2, 3, 4, 5], 2))
+    [[1, 2], [3, 4], [5]]
+    >>> list(chunks([], 2))
+    []
+    >>> list(chunks([1], 2))
+    [[1]]
+    """
+    it = iter(seq)
+    while True:
+        chunk = list(islice(it, n))
+        if not chunk:
+            break
+        yield chunk
+
+
+def type_has(t, required_fields):
+    """Check whether a complex dataType has specific fields.
+
+    Args:
+        t (pyspark.sql.types.ArrayType, pyspark.sql.types.MapType or
+            pyspark.sql.types.StructType): type to check.
+        required_fields (same with t or dict[str: pyspark.sql.DataType]):
+            fields that need to be present in t. For convenience, a user
+            can define a dict in place of a pyspark.sql.types.StructType,
+            but other than that this argument must have the same type as
+            t.
+
+    Raises:
+        AssertionError, if t and required_fields cannot be compared
+           because they aren't instances of the same complex dataType.
+        KeyError, if a required field is not found in the struct.
```

Comment:
I am not sure about using default exceptions here. Two problems:
- hard to understand when something goes wrong in the function itself
- hard to provide a good feedback on the error